src/pages/PreChoice.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button, Input } from "@nextui-org/react";
+import { useState } from "react";
+
+const API_URL = "http://localhost:8000";
+
+export const PreChoice = () => {
+  let [lat, setLat] = useState(0);
+  let [long, setLong] = useState(0);
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      const latitude = position.coords.latitude;
+      const longitude = position.coords.longitude;
+      setLat(latitude);
+      setLong(longitude);
+      console.log(`Latitude: ${latitude}, Longitude: ${longitude}`);
+    });
+  } else {
+    console.log("Geolocation is not supported by this browser.");
+  }
+  let sendCoords = async () => {
+    //placeholder
+    console.log("sending coordinates to backend...")
+  };
+  return (
+    <div
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        justifyContent: "center",
+      }}
+    >
+      {lat ? (
+        <p>
+          Looking for restaurants nearby......
+          <br />
+          Your coordinates are:
+          <br />
+          {`Latitude: ${lat}, Longitude: ${long}`}
+          <br />
+          <Button onClick={() => sendCoords()}>OK</Button>
+        </p>
+      ) : (
+        <div>
```

Comment:
do we want to have users input restaurants if they cannot find them? or can we do a sort of thing where we can maybe provide like a list of common chain restaurants and they can select the ones they know are near them (like checkboxes)